### PR TITLE
PR for #370 - Tab event

### PR DESCRIPTION
**Pull request by angelo510**
Make Tab key to select item as well as add items

### CODE REVIEW

File: src/ng-select/ng-select.component.ts [hunk 1]
```diff
@@ -592,7 +592,12 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
 
     private _handleTab(_: KeyboardEvent) {
         if (this.isOpen) {
-            this.close();
+            if (this.itemsList.markedItem) {
```
Comment (anjmao) [inline src/ng-select/ng-select.component.ts]: Lets make it configurable, you can add new input `tabSelectsValue` which by default can be true so tab will select value by default.

Comment (varnastadeus) [reply src/ng-select/ng-select.component.ts]: I would suggest `selectOnTab`

Comment (anjmao) [reply src/ng-select/ng-select.component.ts]: @varnastadeus i agree

**Comment by anjmao** [comment]
Will be merged in separate PR https://github.com/ng-select/ng-select/pull/397

**Pull request closed without merging**